### PR TITLE
Feature/270 full dv guidance scenario

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** Confluence 270
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

A BSK-Sim scenario was created that illustrates the the usage of the Lambert modules for the first and second maneuver of the Lambert transfer arc. The LambertPlanner and LambertValidator modules were refactored slightly to prevent unnecessary BSK_ERROR messages for other BSK-Sim scenarios that automatically create the Lambert modules but do not configure them because they are not used.
The commits are organized the following way:

- Commits 1: Refactor LambertPlanner and LambertValidator
- Commit 2-3: Update BSK-Sim support files (BSK_Fsw and BSK_Plotting)
- Commit 4: Add BSK-Sim scenario
- Commits 5: Update BSK release notes

## Verification

This scenario combines and illustrates the five modules lambertPlanner, lambertSolver, lambertValidator, lambertSurfaceRelativeVelocity and lambertSecondDV, all of which include their own UnitTest. This BSK-Sim scenario only connects these modules, so no test is included.

## Documentation

Nothing was invalidated.

## Future work

None.

### CODE REVIEW

File: examples/BskSim/models/BSK_Fsw.py [hunk 1]
```diff
@@ -330,6 +364,46 @@ def SetRWMotorTorque(self):
         messaging.ArrayMotorTorqueMsg_C_addAuthor(self.rwMotorTorqueData.rwMotorTorqueOutMsg, self.cmdRwMotorMsg)
         self.rwMotorTorqueData.rwParamsInMsg.subscribeTo(self.fswRwConfigMsg)
 
+
+    def SetLambertPlannerObject(self, SimBase):
+        """Set the lambert planner object."""
```
Comment (thibaudteil) [inline examples/BskSim/models/BSK_Fsw.py:368]: We usually add the modelTag near the module instantiation higher in the file, any specific reason to having it here?

File: examples/BskSim/models/BSK_Fsw.py [hunk 2]
```diff
@@ -330,6 +364,46 @@ def SetRWMotorTorque(self):
...
+            SimBase.DynModels.EarthEphemObject.ephemOutMsgs[0])
+
+
+    def SetLambertSecondDvObject(self):
+        """Set the lambert second DV object."""
+        self.lambertSecondDvObject.ModelTag = "LambertSecondDV"
+        self.lambertSecondDvObject.lambertSolutionInMsg.subscribeTo(self.lambertSolverObject.lambertSolutionOutMsg)
+        self.lambertSecondDvObject.desiredVelocityInMsg.subscribeTo(
+            self.lambertSurfaceRelativeVelocityObject.desiredVelocityOutMsg)
+
```
Comment (thibaudteil) [inline examples/BskSim/models/BSK_Fsw.py]: Same comment for the ModelTags here

File: examples/BskSim/scenarios/scenario_LambertGuidance.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    TheScenario = scenario_LambertGuidance()
+    runScenario(TheScenario)
+    figureList = TheScenario.pull_outputs(showPlots)
+
+    return figureList
+
+
+
+
+
```
Comment (thibaudteil) [inline examples/BskSim/scenarios/scenario_LambertGuidance.py]: A couple extra lines here

**Review by thibaudteil — COMMENTED** [review]
Good work!

File: examples/BskSim/models/BSK_Fsw.py (hunk 1 above)
Comment (JulianHammerl) [reply examples/BskSim/models/BSK_Fsw.py:368]: I looked at how C++ modules were configured in the Dynamics file, and this is how it was done there. I did change it now in the FSW file to be consistent.

File: examples/BskSim/models/BSK_Fsw.py (hunk 2 above)
Comment (JulianHammerl) [reply examples/BskSim/models/BSK_Fsw.py]: Fixed.

File: examples/BskSim/scenarios/scenario_LambertGuidance.py (hunk 3 above)
Comment (JulianHammerl) [reply examples/BskSim/scenarios/scenario_LambertGuidance.py]: Fixed.

**Review by thibaudteil — APPROVED** [review]

**Pull request merged**